rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake

```diff
@@ -177,105 +176,94 @@ else()
-    find_package(${_typesupport_impl} REQUIRED)
-    set(_generated_msg_c_common_file "${_full_folder}/${_base_msg_name}.c")
-
-    set(_pyext_suffix "__pyext")
-
-    # TODO(esteve): Change the following code so that each file is compiled independently
-    add_library(${_msg_name}${_pyext_suffix} SHARED
-      "${_generated_msg_c_ts_file}"
-      "${_generated_msg_c_common_file}"
-    )
+  set(_pyext_suffix "__pyext")
+  set(_target_name "${PROJECT_NAME}${_typesupport_impl}${_pyext_suffix}")
 
-    set_properties("")
-    if(WIN32)
-      set_properties("_DEBUG")
-      set_properties("_MINSIZEREL")
-      set_properties("_RELEASE")
-      set_properties("_RELWITHDEBINFO")
-    endif()
-
-    add_dependencies(
-      ${_msg_name}${_pyext_suffix}
-      ${rosidl_generate_interfaces_TARGET}${_target_suffix}
-      ${rosidl_generate_interfaces_TARGET}__rosidl_generator_c
-    )
+  add_library(${_target_name} SHARED
+    ${_generated_msg_c_ts_${_typesupport_impl}_files}
+    ${_generated_msg_c_common_files}
+  )
 
-    set(_extension_compile_flags "")
-    if(NOT WIN32)
-      set(_extension_compile_flags "-Wall -Wextra")
-    endif()
+  add_dependencies(
+    ${_target_name}
+    ${rosidl_generate_interfaces_TARGET}${_target_suffix}
+    ${rosidl_generate_interfaces_TARGET}__rosidl_generator_c
+  )
 
-    target_link_libraries(
-      ${_msg_name}${_pyext_suffix}
-      ${PythonExtra_LIBRARIES}
-      ${PROJECT_NAME}__${_typesupport_impl}
-    )
+  set(_extension_compile_flags "")
+  if(NOT WIN32)
+    set(_extension_compile_flags "-Wall -Wextra")
+  endif()
+  set_properties("")
+  if(WIN32)
+    set_properties("_DEBUG")
+    set_properties("_MINSIZEREL")
+    set_properties("_RELEASE")
+    set_properties("_RELWITHDEBINFO")
+  endif()
+  target_link_libraries(
+    ${_target_name}
+    ${PythonExtra_LIBRARIES}
+    ${PROJECT_NAME}__${_typesupport_impl}
+  )
 
-    list(APPEND _generated_extension_files
-      "$<TARGET_FILE:${_msg_name}${_pyext_suffix}>"
-    )
+  list(APPEND _generated_extension_files
+    "$<TARGET_FILE:${_target_name}>"
+  )
 
-    target_include_directories(${_msg_name}${_pyext_suffix}
-      PUBLIC
-      ${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_c
-      ${PythonExtra_INCLUDE_DIRS}
-    )
+  target_include_directories(${_target_name}
+    PUBLIC
+    ${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_c
+    ${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_py
+    ${PythonExtra_INCLUDE_DIRS}
+  )
 
-    foreach(_pkg_name ${rosidl_generate_interfaces_DEPENDENCY_PACKAGE_NAMES})
-      ament_target_dependencies(
-        ${_msg_name}${_pyext_suffix}
-        ${_pkg_name}
-      )
-    endforeach()
-    ament_target_dependencies(${_msg_name}${_pyext_suffix}
-      "rosidl_generator_c"
-      "${_typesupport_impl}"
+  foreach(_pkg_name ${rosidl_generate_interfaces_DEPENDENCY_PACKAGE_NAMES})
+    ament_target_dependencies(${_target_name}
+      ${_pkg_name}
+      "_${_pkg_name}__${_typesupport_impl}${PythonExtra_EXTENSION_SUFFIX}"
```

Comment:
Please try to add debug message in there to check why you don't get an error message. It would be good to ensure that the API works as intended even if you don't need it for this change.